src/TorchSharp/Tensor/Storage.cs

```diff
@@ -0,0 +1,522 @@
+            /// </summary>
+            public Storage<int> @int() => _tensor.to_type(ScalarType.Int32).storage<int>();
+
+            /// <summary>
+            /// Convert to long storage.
+            /// </summary>
+            public Storage<long> @long() => _tensor.to_type(ScalarType.Int64).storage<long>();
+
+            /// <summary>
+            /// Convert to float storage.
+            /// </summary>
+            public Storage<float> @float() => _tensor.to_type(ScalarType.Float32).storage<float>();
+
+            /// <summary>
+            /// Convert to double storage.
+            /// </summary>
+            public Storage<double> @double() => _tensor.to_type(ScalarType.Float64).storage<double>();
+
+            /// <summary>
+            /// Convert to 32-bit complex storage.
+            /// </summary>
+            public Storage<(float,float)> complex_float() => _tensor.to_type(ScalarType.ComplexFloat32).storage<(float, float)>();
+
+            /// <summary>
+            /// Convert to 64-bit complex storage.
+            /// </summary>
+            public Storage<System.Numerics.Complex> complex_double() => _tensor.to_type(ScalarType.ComplexFloat64).storage<System.Numerics.Complex>();
+
+            /// <summary>
+            /// The size of each storage element.
+            /// </summary>
+            /// <returns></returns>
+            public abstract int element_size();
+
+            /// <summary>
+            /// A pointer to the raw data in memory.
+            /// </summary>
+            /// <returns></returns>
+            protected IntPtr data_ptr()
+            {
+                if (_tensor_data_ptr != IntPtr.Zero)
+                    return _tensor_data_ptr;
+
+                var res = THSStorage_data_ptr(_tensor.Handle);
+                if (res == IntPtr.Zero) { CheckForErrors(); }
+                _tensor_data_ptr = res;
+                return res;
+            }
+
+            /// <summary>
+            /// The number of bytes allocated to the storage.
+            /// </summary>
+            /// <returns></returns>
+            public UInt64 nbytes()
+            {
+                var res = THSStorage_nbytes(_tensor.Handle);
+                CheckForErrors();
+                return res;
+            }
+
+            [DllImport("LibTorchSharp")]
+            protected static extern UInt64 THSStorage_nbytes(IntPtr tensor);
+            [DllImport("LibTorchSharp")]
+            protected static extern void THSStorage_set_nbytes(IntPtr tensor, UInt64 nbytes);
+            [DllImport("LibTorchSharp")]
+            protected static extern IntPtr THSStorage_data_ptr(IntPtr tensor);
+        }
+
+        /// <summary>
+        /// A torch.Storage is a contiguous, one-dimensional array of a single data type.
+        /// Every tensor has a corresponding storage of the same data type.
+        /// </summary>
+        public class Storage<T> : torch.Storage, IDisposable, IEnumerable<T> where T : unmanaged
```

Comment:
>class [](http://example.com/codeflow?start=15&length=5)

Mark this as sealed if we think no-one want to subclass it. This will have some perf enhancement when it is sealed.